Sources/TuistKit/Services/Migration/MigrationTargetsByDependenciesService.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import TSCBasic
+import TuistMigration
+import TuistSupport
+
+final class MigrationTargetsByDependenciesService {
+    // MARK: - Attributes
+
+    private let targetsExtractor: TargetsExtracting
+
+    // MARK: - Init
+
+    init(targetsExtractor: TargetsExtracting = TargetsExtractor()) {
+        self.targetsExtractor = targetsExtractor
+    }
+
+    // MARK: - Internal
+
+    func run(xcodeprojPath: AbsolutePath) throws {
```

Comment:
@andreacipriani what about adding support for `--json`. If the argument is passed, we can ouptut a JSON instead:

```json
{
  "targetA": 33,
  "tarbetB": 44
}
```

Reply:
Do you think about adding JSON only to simplify acceptance testing or because users would need? I feel this command is only informative and we shouldn't over complicate it.
I added one acc test directly on the output, wdyt?

Reply:
I think is more about having acceptance tests that are easy to maintain. Acceptance tests are generally costly to maintain. If we run assertions on the output, any change in the syntax will cause the tests to fail, which is not ideal because we'll waste time that we'd otherwise invest elsewhere.

Reply:
I made json the default output, I don't think we need multiple ways to format it. I've updated the tests consequentially